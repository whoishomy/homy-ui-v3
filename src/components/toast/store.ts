import { create } from 'zustand';

export interface Toast {
  id: string;
  message: string;
  type?: 'success' | 'error' | 'warning' | 'info';
  duration?: number;
}

interface ToastStore {
  toasts: Toast[];
  add: (toast: Toast) => void;
  remove: (id: string) => void;
  update: (id: string, toast: Partial<Toast>) => void;
  clear: () => void;
}

export const toastStore = create<ToastStore>((set) => ({
  toasts: [],
  add: (toast) => set((state) => ({ toasts: [...state.toasts, toast] })),
  remove: (id) => set((state) => ({ toasts: state.toasts.filter((t) => t.id !== id) })),
  update: (id, toast) =>
    set((state) => ({
      toasts: state.toasts.map((t) => (t.id === id ? { ...t, ...toast } : t)),
    })),
  clear: () => set({ toasts: [] }),
})); 