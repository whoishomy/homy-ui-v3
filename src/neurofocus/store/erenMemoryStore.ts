import { NeuroFocusMemory, EmotionalResponse, TaskBreakdown } from '../core/types';

export interface ErenSnapshot {
  timestamp: string;
  emotionalState: EmotionalResponse;
  achievement?: {
    title: string;
    description: string;
    difficulty: 'easy' | 'medium' | 'hard';
    completionTime: number;
  };
  supportMessage?: string;
}

export interface ErenProgress {
  dailyEmotionalJourney: ErenSnapshot[];
  completedTasks: number;
  totalFocusTime: number;
  breakthroughs: Array<{
    date: string;
    description: string;
    emotionalImpact: string;
  }>;
  supportHistory: Array<{
    trigger: string;
    response: string;
    effectiveness: number;
  }>;
}

export class ErenMemoryStore {
  private memories: NeuroFocusMemory[] = [];
  private progress: ErenProgress = {
    dailyEmotionalJourney: [],
    completedTasks: 0,
    totalFocusTime: 0,
    breakthroughs: [],
    supportHistory: [],
  };

  async recordEmotionalSnapshot(
    emotion: EmotionalResponse,
    context?: { taskId: string; supportMessage?: string }
  ): Promise<void> {
    const snapshot: ErenSnapshot = {
      timestamp: new Date().toISOString(),
      emotionalState: emotion,
      supportMessage: context?.supportMessage,
    };

    this.progress.dailyEmotionalJourney.push(snapshot);

    // Emit real-time event for UI updates
    this.emitProgressUpdate('emotional-snapshot', snapshot);
  }

  async recordBreakthrough(description: string, emotionalImpact: string): Promise<void> {
    const breakthrough = {
      date: new Date().toISOString(),
      description,
      emotionalImpact,
    };

    this.progress.breakthroughs.push(breakthrough);
    this.emitProgressUpdate('breakthrough', breakthrough);
  }

  async recordSupportInteraction(
    trigger: string,
    response: string,
    effectiveness: number
  ): Promise<void> {
    const interaction = {
      trigger,
      response,
      effectiveness,
    };

    this.progress.supportHistory.push(interaction);
    this.emitProgressUpdate('support-interaction', interaction);
  }

  async getDailyProgress(): Promise<{
    dominantEmotion: string;
    improvements: string[];
    supportEffectiveness: number;
  }> {
    const today = new Date().toISOString().split('T')[0];
    const todaySnapshots = this.progress.dailyEmotionalJourney.filter((snapshot) =>
      snapshot.timestamp.startsWith(today)
    );

    // Analyze emotional journey
    const emotions = todaySnapshots.map((s) => s.emotionalState.type);
    const dominantEmotion = this.findDominantEmotion(emotions);

    // Calculate support effectiveness
    const todaySupport = this.progress.supportHistory.filter((s) => s.trigger.startsWith(today));
    const avgEffectiveness =
      todaySupport.reduce((sum, s) => sum + s.effectiveness, 0) / (todaySupport.length || 1);

    return {
      dominantEmotion,
      improvements: this.identifyImprovements(todaySnapshots),
      supportEffectiveness: avgEffectiveness,
    };
  }

  private findDominantEmotion(emotions: string[]): string {
    const counts = emotions.reduce(
      (acc, emotion) => {
        acc[emotion] = (acc[emotion] || 0) + 1;
        return acc;
      },
      {} as Record<string, number>
    );

    return Object.entries(counts).sort((a, b) => b[1] - a[1])[0]?.[0] || 'neutral';
  }

  private identifyImprovements(snapshots: ErenSnapshot[]): string[] {
    const improvements: string[] = [];

    // Analyze emotional transitions
    for (let i = 1; i < snapshots.length; i++) {
      const prev = snapshots[i - 1].emotionalState.type;
      const curr = snapshots[i].emotionalState.type;

      if (
        (prev === 'overwhelmed' && curr === 'determined') ||
        (prev === 'frustrated' && curr === 'curious') ||
        (prev === 'tired' && curr === 'exploring')
      ) {
        improvements.push(
          `Duygusal geçiş: ${prev} durumundan ${curr} durumuna ilerleme kaydedildi.`
        );
      }
    }

    return improvements;
  }

  private emitProgressUpdate(type: string, data: any): void {
    // Event emitter implementation for real-time updates
    window.dispatchEvent(
      new CustomEvent('eren-progress', {
        detail: { type, data },
      })
    );
  }
}

// Export singleton instance
export const erenMemoryStore = new ErenMemoryStore();
