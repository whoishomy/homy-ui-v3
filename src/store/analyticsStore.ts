import { create } from 'zustand';
import { devtools } from 'zustand/middleware';
import {
  AnalyticsState,
  AnalyticsFilters,
  AnalyticsSummary,
  HealthInsight,
} from '@/types/analytics';
import { analyticsService } from '@/services/analyticsService';

interface AnalyticsStore extends AnalyticsState {
  fetchSummary: (filters: AnalyticsFilters) => Promise<void>;
  fetchInsights: (filters: AnalyticsFilters) => Promise<void>;
  setFilters: (filters: AnalyticsFilters) => void;
  reset: () => void;
}

const initialState: AnalyticsState = {
  summary: null,
  insights: [],
  loading: false,
  error: null,
  filters: {
    timeRange: 'weekly',
    onlyActive: true,
    startDate: (() => {
      const now = new Date();
      const startDate = new Date(now);
      startDate.setDate(now.getDate() - 7); // Default to last 7 days
      return startDate;
    })(),
    endDate: new Date(),
  },
};

const useAnalyticsStore = create<AnalyticsStore>()(
  devtools(
    (set, get) => ({
      ...initialState,

      fetchSummary: async (filters: AnalyticsFilters) => {
        try {
          set({ loading: true, error: null });
          const summary = await analyticsService.getSummary(filters);
          set({ summary, loading: false });
        } catch (error) {
          set({
            error: error instanceof Error ? error.message : 'Veri yüklenirken bir hata oluştu',
            loading: false,
          });
        }
      },

      fetchInsights: async (filters: AnalyticsFilters) => {
        try {
          set({ loading: true, error: null });
          const insights = await analyticsService.getInsights(filters);
          set({ insights, loading: false });
        } catch (error) {
          set({
            error: error instanceof Error ? error.message : 'İçgörüler yüklenirken bir hata oluştu',
            loading: false,
          });
        }
      },

      setFilters: (filters: AnalyticsFilters) => {
        set({ filters });
      },

      reset: () => {
        set(initialState);
      },
    }),
    {
      name: 'analytics-store',
    }
  )
);

export { useAnalyticsStore };
export type { AnalyticsStore }; 