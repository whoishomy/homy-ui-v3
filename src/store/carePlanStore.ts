import { create } from 'zustand';
import { devtools, persist } from 'zustand/middleware';
import { v4 as uuidv4 } from 'uuid';
import {
  CarePlan,
  Medication,
  Appointment,
  HealthGoal,
  HealthMetric,
  carePlanSchema,
} from '../types/carePlan';

interface CarePlanState {
  carePlans: CarePlan[];
  activePlanId: string | null;
  isLoading: boolean;
  error: string | null;

  // CarePlan CRUD
  createCarePlan: (plan: Omit<CarePlan, 'id' | 'createdAt' | 'updatedAt'>) => Promise<CarePlan>;
  updateCarePlan: (id: string, plan: Partial<CarePlan>) => Promise<CarePlan>;
  deleteCarePlan: (id: string) => Promise<void>;
  setActivePlan: (id: string | null) => void;

  // Medication Management
  addMedication: (planId: string, medication: Omit<Medication, 'id'>) => Promise<void>;
  updateMedication: (planId: string, medicationId: string, updates: Partial<Medication>) => Promise<void>;
  deleteMedication: (planId: string, medicationId: string) => Promise<void>;

  // Appointment Management
  addAppointment: (planId: string, appointment: Omit<Appointment, 'id'>) => Promise<void>;
  updateAppointment: (planId: string, appointmentId: string, updates: Partial<Appointment>) => Promise<void>;
  deleteAppointment: (planId: string, appointmentId: string) => Promise<void>;

  // Health Goal Management
  addHealthGoal: (planId: string, goal: Omit<HealthGoal, 'id'>) => Promise<void>;
  updateHealthGoal: (planId: string, goalId: string, updates: Partial<HealthGoal>) => Promise<void>;
  deleteHealthGoal: (planId: string, goalId: string) => Promise<void>;

  // Health Metric Management
  addHealthMetric: (planId: string, metric: Omit<HealthMetric, 'id'>) => Promise<void>;
  updateHealthMetric: (planId: string, metricId: string, updates: Partial<HealthMetric>) => Promise<void>;
  deleteHealthMetric: (planId: string, metricId: string) => Promise<void>;
}

export const useCarePlanStore = create<CarePlanState>()(
  devtools(
    persist(
      (set, get) => ({
        carePlans: [],
        activePlanId: null,
        isLoading: false,
        error: null,

        // CarePlan CRUD
        createCarePlan: async (planData) => {
          set({ isLoading: true, error: null });
          try {
            const newPlan: CarePlan = {
              ...planData,
              id: uuidv4(),
              createdAt: new Date(),
              updatedAt: new Date(),
            };
            
            // Validate with Zod schema
            carePlanSchema.parse(newPlan);
            
            set((state) => ({
              carePlans: [...state.carePlans, newPlan],
              activePlanId: newPlan.id,
            }));
            return newPlan;
          } catch (error) {
            const errorMessage = error instanceof Error ? error.message : 'Bakım planı oluşturulamadı';
            set({ error: errorMessage });
            throw error;
          } finally {
            set({ isLoading: false });
          }
        },

        updateCarePlan: async (id, updates) => {
          set({ isLoading: true, error: null });
          try {
            const state = get();
            const planIndex = state.carePlans.findIndex((plan) => plan.id === id);
            
            if (planIndex === -1) {
              throw new Error('Bakım planı bulunamadı');
            }

            const updatedPlan: CarePlan = {
              ...state.carePlans[planIndex],
              ...updates,
              updatedAt: new Date(),
            };

            // Validate with Zod schema
            carePlanSchema.parse(updatedPlan);

            const newCarePlans = [...state.carePlans];
            newCarePlans[planIndex] = updatedPlan;

            set({ carePlans: newCarePlans });
            return updatedPlan;
          } catch (error) {
            const errorMessage = error instanceof Error ? error.message : 'Bakım planı güncellenemedi';
            set({ error: errorMessage });
            throw error;
          } finally {
            set({ isLoading: false });
          }
        },

        deleteCarePlan: async (id) => {
          set({ isLoading: true, error: null });
          try {
            set((state) => ({
              carePlans: state.carePlans.filter((plan) => plan.id !== id),
              activePlanId: state.activePlanId === id ? null : state.activePlanId,
            }));
          } catch (error) {
            const errorMessage = error instanceof Error ? error.message : 'Bakım planı silinemedi';
            set({ error: errorMessage });
            throw error;
          } finally {
            set({ isLoading: false });
          }
        },

        setActivePlan: (id) => {
          set({ activePlanId: id });
        },

        // Medication Management
        addMedication: async (planId, medication) => {
          set({ isLoading: true, error: null });
          try {
            const newMedication: Medication = {
              ...medication,
              id: uuidv4(),
            };

            set((state) => ({
              carePlans: state.carePlans.map((plan) =>
                plan.id === planId
                  ? { ...plan, medications: [...plan.medications, newMedication], updatedAt: new Date() }
                  : plan
              ),
            }));
          } catch (error) {
            const errorMessage = error instanceof Error ? error.message : 'İlaç eklenemedi';
            set({ error: errorMessage });
            throw error;
          } finally {
            set({ isLoading: false });
          }
        },

        updateMedication: async (planId, medicationId, updates) => {
          set({ isLoading: true, error: null });
          try {
            set((state) => ({
              carePlans: state.carePlans.map((plan) =>
                plan.id === planId
                  ? {
                      ...plan,
                      medications: plan.medications.map((med) =>
                        med.id === medicationId ? { ...med, ...updates } : med
                      ),
                      updatedAt: new Date(),
                    }
                  : plan
              ),
            }));
          } catch (error) {
            const errorMessage = error instanceof Error ? error.message : 'İlaç güncellenemedi';
            set({ error: errorMessage });
            throw error;
          } finally {
            set({ isLoading: false });
          }
        },

        deleteMedication: async (planId, medicationId) => {
          set({ isLoading: true, error: null });
          try {
            set((state) => ({
              carePlans: state.carePlans.map((plan) =>
                plan.id === planId
                  ? {
                      ...plan,
                      medications: plan.medications.filter((med) => med.id !== medicationId),
                      updatedAt: new Date(),
                    }
                  : plan
              ),
            }));
          } catch (error) {
            const errorMessage = error instanceof Error ? error.message : 'İlaç silinemedi';
            set({ error: errorMessage });
            throw error;
          } finally {
            set({ isLoading: false });
          }
        },

        // Appointment Management
        addAppointment: async (planId, appointment) => {
          set({ isLoading: true, error: null });
          try {
            const newAppointment: Appointment = {
              ...appointment,
              id: uuidv4(),
            };

            set((state) => ({
              carePlans: state.carePlans.map((plan) =>
                plan.id === planId
                  ? { ...plan, appointments: [...plan.appointments, newAppointment], updatedAt: new Date() }
                  : plan
              ),
            }));
          } catch (error) {
            const errorMessage = error instanceof Error ? error.message : 'Randevu eklenemedi';
            set({ error: errorMessage });
            throw error;
          } finally {
            set({ isLoading: false });
          }
        },

        updateAppointment: async (planId, appointmentId, updates) => {
          set({ isLoading: true, error: null });
          try {
            set((state) => ({
              carePlans: state.carePlans.map((plan) =>
                plan.id === planId
                  ? {
                      ...plan,
                      appointments: plan.appointments.map((apt) =>
                        apt.id === appointmentId ? { ...apt, ...updates } : apt
                      ),
                      updatedAt: new Date(),
                    }
                  : plan
              ),
            }));
          } catch (error) {
            const errorMessage = error instanceof Error ? error.message : 'Randevu güncellenemedi';
            set({ error: errorMessage });
            throw error;
          } finally {
            set({ isLoading: false });
          }
        },

        deleteAppointment: async (planId, appointmentId) => {
          set({ isLoading: true, error: null });
          try {
            set((state) => ({
              carePlans: state.carePlans.map((plan) =>
                plan.id === planId
                  ? {
                      ...plan,
                      appointments: plan.appointments.filter((apt) => apt.id !== appointmentId),
                      updatedAt: new Date(),
                    }
                  : plan
              ),
            }));
          } catch (error) {
            const errorMessage = error instanceof Error ? error.message : 'Randevu silinemedi';
            set({ error: errorMessage });
            throw error;
          } finally {
            set({ isLoading: false });
          }
        },

        // Health Goal Management
        addHealthGoal: async (planId, goal) => {
          set({ isLoading: true, error: null });
          try {
            const newGoal: HealthGoal = {
              ...goal,
              id: uuidv4(),
            };

            set((state) => ({
              carePlans: state.carePlans.map((plan) =>
                plan.id === planId
                  ? { ...plan, goals: [...plan.goals, newGoal], updatedAt: new Date() }
                  : plan
              ),
            }));
          } catch (error) {
            const errorMessage = error instanceof Error ? error.message : 'Hedef eklenemedi';
            set({ error: errorMessage });
            throw error;
          } finally {
            set({ isLoading: false });
          }
        },

        updateHealthGoal: async (planId, goalId, updates) => {
          set({ isLoading: true, error: null });
          try {
            set((state) => ({
              carePlans: state.carePlans.map((plan) =>
                plan.id === planId
                  ? {
                      ...plan,
                      goals: plan.goals.map((goal) =>
                        goal.id === goalId ? { ...goal, ...updates } : goal
                      ),
                      updatedAt: new Date(),
                    }
                  : plan
              ),
            }));
          } catch (error) {
            const errorMessage = error instanceof Error ? error.message : 'Hedef güncellenemedi';
            set({ error: errorMessage });
            throw error;
          } finally {
            set({ isLoading: false });
          }
        },

        deleteHealthGoal: async (planId, goalId) => {
          set({ isLoading: true, error: null });
          try {
            set((state) => ({
              carePlans: state.carePlans.map((plan) =>
                plan.id === planId
                  ? {
                      ...plan,
                      goals: plan.goals.filter((goal) => goal.id !== goalId),
                      updatedAt: new Date(),
                    }
                  : plan
              ),
            }));
          } catch (error) {
            const errorMessage = error instanceof Error ? error.message : 'Hedef silinemedi';
            set({ error: errorMessage });
            throw error;
          } finally {
            set({ isLoading: false });
          }
        },

        // Health Metric Management
        addHealthMetric: async (planId, metric) => {
          set({ isLoading: true, error: null });
          try {
            const newMetric: HealthMetric = {
              ...metric,
              id: uuidv4(),
            };

            set((state) => ({
              carePlans: state.carePlans.map((plan) =>
                plan.id === planId
                  ? { ...plan, metrics: [...plan.metrics, newMetric], updatedAt: new Date() }
                  : plan
              ),
            }));
          } catch (error) {
            const errorMessage = error instanceof Error ? error.message : 'Metrik eklenemedi';
            set({ error: errorMessage });
            throw error;
          } finally {
            set({ isLoading: false });
          }
        },

        updateHealthMetric: async (planId, metricId, updates) => {
          set({ isLoading: true, error: null });
          try {
            set((state) => ({
              carePlans: state.carePlans.map((plan) =>
                plan.id === planId
                  ? {
                      ...plan,
                      metrics: plan.metrics.map((metric) =>
                        metric.id === metricId ? { ...metric, ...updates } : metric
                      ),
                      updatedAt: new Date(),
                    }
                  : plan
              ),
            }));
          } catch (error) {
            const errorMessage = error instanceof Error ? error.message : 'Metrik güncellenemedi';
            set({ error: errorMessage });
            throw error;
          } finally {
            set({ isLoading: false });
          }
        },

        deleteHealthMetric: async (planId, metricId) => {
          set({ isLoading: true, error: null });
          try {
            set((state) => ({
              carePlans: state.carePlans.map((plan) =>
                plan.id === planId
                  ? {
                      ...plan,
                      metrics: plan.metrics.filter((metric) => metric.id !== metricId),
                      updatedAt: new Date(),
                    }
                  : plan
              ),
            }));
          } catch (error) {
            const errorMessage = error instanceof Error ? error.message : 'Metrik silinemedi';
            set({ error: errorMessage });
            throw error;
          } finally {
            set({ isLoading: false });
          }
        },
      }),
      {
        name: 'care-plan-storage',
        version: 1,
      }
    )
  )
); 