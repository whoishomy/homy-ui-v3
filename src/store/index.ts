export * from './analyticsStore';
export * from './carePlanStore'; 