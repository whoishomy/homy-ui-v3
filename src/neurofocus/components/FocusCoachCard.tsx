import React, { useEffect, useState } from 'react';
import { motion, AnimatePresence } from 'framer-motion';
import { Card } from '@/components/ui/Card';
import { Badge } from '@/components/ui/Badge';
import { erenMemoryStore, ErenSnapshot } from '../store/erenMemoryStore';
import { EmotionalResponse } from '../core/types';

interface FocusCoachCardProps {
  currentTask?: string;
  onEmotionChange?: (emotion: EmotionalResponse) => void;
}

export const FocusCoachCard: React.FC<FocusCoachCardProps> = ({ currentTask, onEmotionChange }) => {
  const [currentSnapshot, setCurrentSnapshot] = useState<ErenSnapshot | null>(null);
  const [showCelebration, setShowCelebration] = useState(false);

  useEffect(() => {
    // Listen for Eren's progress updates
    const handleProgressUpdate = (event: CustomEvent) => {
      const { type, data } = event.detail;

      if (type === 'emotional-snapshot') {
        setCurrentSnapshot(data);
      } else if (type === 'breakthrough') {
        setShowCelebration(true);
        setTimeout(() => setShowCelebration(false), 3000);
      }
    };

    window.addEventListener('eren-progress' as any, handleProgressUpdate);
    return () => {
      window.removeEventListener('eren-progress' as any, handleProgressUpdate);
    };
  }, []);

  const renderEmotionalSupport = () => {
    if (!currentSnapshot) return null;

    const { emotionalState } = currentSnapshot;

    return (
      <div className="space-y-4">
        <div className="flex items-center gap-2">
          <Badge variant={emotionalState.type === 'determined' ? 'success' : 'default'}>
            {emotionalState.type}
          </Badge>
          <span className="text-sm text-gray-600">
            {new Date(currentSnapshot.timestamp).toLocaleTimeString('tr-TR')}
          </span>
        </div>

        <p className="text-lg font-medium">
          {emotionalState.type === 'overwhelmed' &&
            'Eren, adım adım ilerleyelim. Her küçük başarı değerli! 🌱'}
          {emotionalState.type === 'frustrated' &&
            'Bazen takılmak normaldir. Birlikte çözüm bulacağız! 💪'}
          {emotionalState.type === 'determined' &&
            'Harika gidiyorsun Eren! Bu motivasyonunu takdir ediyorum! ⭐'}
          {emotionalState.type === 'curious' &&
            'Merakın en büyük süper gücün! Keşfetmeye devam et! 🔍'}
          {emotionalState.type === 'tired' &&
            'Kısa bir mola vermek ister misin? Dinlenmek de başarının bir parçası! 🌿'}
        </p>

        {currentSnapshot.supportMessage && (
          <div className="mt-4 p-4 bg-blue-50 rounded-lg">
            <p className="text-blue-800">{currentSnapshot.supportMessage}</p>
          </div>
        )}
      </div>
    );
  };

  const renderCelebration = () => {
    return (
      <AnimatePresence>
        {showCelebration && (
          <motion.div
            initial={{ scale: 0, opacity: 0 }}
            animate={{ scale: 1, opacity: 1 }}
            exit={{ scale: 0, opacity: 0 }}
            className="absolute inset-0 flex items-center justify-center bg-blue-500/20 backdrop-blur-sm rounded-lg"
          >
            <motion.div
              animate={{ y: [0, -20, 0] }}
              transition={{ repeat: 3, duration: 0.5 }}
              className="text-4xl"
            >
              🌟
            </motion.div>
            <p className="text-xl font-bold text-blue-800 ml-4">Harika bir gelişme kaydedildi!</p>
          </motion.div>
        )}
      </AnimatePresence>
    );
  };

  return (
    <Card className="relative overflow-hidden p-6 bg-gradient-to-br from-blue-50 to-white">
      <div className="flex items-center justify-between mb-6">
        <h3 className="text-xl font-semibold text-blue-900">Eren'in Koçu</h3>
        {currentTask && (
          <Badge variant="outline" className="text-sm">
            {currentTask}
          </Badge>
        )}
      </div>

      {renderEmotionalSupport()}
      {renderCelebration()}

      <div className="mt-6 grid grid-cols-2 gap-2">
        <button
          onClick={() => {
            const emotion: EmotionalResponse = {
              type: 'determined',
              intensity: 0.8,
              timestamp: new Date().toISOString(),
            };
            erenMemoryStore.recordEmotionalSnapshot(emotion);
            onEmotionChange?.(emotion);
          }}
          className="px-4 py-2 bg-green-100 text-green-700 rounded-lg hover:bg-green-200 transition-colors"
        >
          Yapabilirim! 💪
        </button>
        <button
          onClick={() => {
            const emotion: EmotionalResponse = {
              type: 'overwhelmed',
              intensity: 0.6,
              timestamp: new Date().toISOString(),
            };
            erenMemoryStore.recordEmotionalSnapshot(emotion);
            onEmotionChange?.(emotion);
          }}
          className="px-4 py-2 bg-blue-100 text-blue-700 rounded-lg hover:bg-blue-200 transition-colors"
        >
          Yardım İstiyorum 🤝
        </button>
      </div>
    </Card>
  );
};
