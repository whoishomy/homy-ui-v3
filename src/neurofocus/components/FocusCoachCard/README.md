# FocusCoachCard – Eren Mode

## 🎯 Amaç

Dikkat dağınıklığı yaşayan çocuklara mikro görevlerle destek sunmak ve duygusal bağlantı kurmak.

## 🌟 Özellikler

### Empati Tabanlı Duygusal Yanıtlar

- Anlık duygu durumu takibi
- Kişiselleştirilmiş destek mesajları
- Pozitif pekiştirme sistemi

### Mikro Görev Sistemi

- Küçük, yönetilebilir adımlar
- İlerleme göstergeleri
- Başarı kutlamaları

### İlerleme Çizgisi (SuccessTimeline)

- Görsel ilerleme takibi
- Motivasyon artırıcı animasyonlar
- Başarı rozetleri

### NeuroFocus Engine Entegrasyonu

- Duygusal analiz
- Adaptif zorluk seviyesi
- Kişiselleştirilmiş öğrenme yolu

## 📱 Kullanım

```tsx
import { FocusCoachCard } from '@/neurofocus/components/FocusCoachCard';

<FocusCoachCard
  currentTask={task}
  onEmotionChange={handleEmotionChange}
  onHelpClick={handleHelpRequest}
  isCompleted={false}
/>;
```

## 🧪 Test Kapsamı

- Görev render testleri
- Duygusal etkileşim testleri
- Erişilebilirlik kontrolleri
- Snapshot karşılaştırmaları
- İlerleme takibi testleri

## 📸 Görsel Örnekler

![Eren Modu](../../../../docs/screenshots/focus-coach/focus-coach-eren-mode.png)

## 🔄 Güncellemeler

- v1.0.0: İlk sürüm
- v1.1.0: Duygusal yanıt sistemi eklendi
- v1.2.0: NeuroFocus Engine entegrasyonu
- v1.3.0: Eren Mode özelleştirmeleri
