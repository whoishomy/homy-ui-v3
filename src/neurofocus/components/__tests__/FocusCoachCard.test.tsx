import React from 'react';
import { render, screen, waitFor, act } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { axe } from 'jest-axe';
import { FocusCoachCard } from '../FocusCoachCard';
import { erenMemoryStore } from '../../store/erenMemoryStore';

// Mock the erenMemoryStore
jest.mock('../../store/erenMemoryStore', () => ({
  erenMemoryStore: {
    recordEmotionalSnapshot: jest.fn(),
  },
}));

describe('FocusCoachCard', () => {
  const mockOnEmotionChange = jest.fn();

  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const { container } = render(
      <FocusCoachCard currentTask="Test Task" onEmotionChange={mockOnEmotionChange} />
    );
    expect(container).toMatchSnapshot();
  });

  it('has no accessibility violations', async () => {
    const { container } = render(
      <FocusCoachCard currentTask="Test Task" onEmotionChange={mockOnEmotionChange} />
    );
    const results = await axe(container);
    expect(results).toHaveNoViolations();
  });

  it('displays task title', () => {
    render(<FocusCoachCard currentTask="Test Task" onEmotionChange={mockOnEmotionChange} />);

    expect(screen.getByText('Test Task')).toBeInTheDocument();
    expect(screen.getByText("Eren'in Koçu")).toBeInTheDocument();
  });

  it('handles determined emotion button click', async () => {
    const user = userEvent.setup();

    render(<FocusCoachCard currentTask="Test Task" onEmotionChange={mockOnEmotionChange} />);

    const determinedButton = screen.getByText('Yapabilirim! 💪');
    await user.click(determinedButton);

    expect(mockOnEmotionChange).toHaveBeenCalledWith(
      expect.objectContaining({
        type: 'determined',
        intensity: 0.8,
      })
    );
    expect(erenMemoryStore.recordEmotionalSnapshot).toHaveBeenCalledWith(
      expect.objectContaining({
        type: 'determined',
        intensity: 0.8,
      })
    );
  });

  it('handles overwhelmed emotion button click', async () => {
    const user = userEvent.setup();

    render(<FocusCoachCard currentTask="Test Task" onEmotionChange={mockOnEmotionChange} />);

    const helpButton = screen.getByText('Yardım İstiyorum 🤝');
    await user.click(helpButton);

    expect(mockOnEmotionChange).toHaveBeenCalledWith(
      expect.objectContaining({
        type: 'overwhelmed',
        intensity: 0.6,
      })
    );
    expect(erenMemoryStore.recordEmotionalSnapshot).toHaveBeenCalledWith(
      expect.objectContaining({
        type: 'overwhelmed',
        intensity: 0.6,
      })
    );
  });

  describe('Visual Regression Tests', () => {
    const viewports = [
      { width: 375, height: 667, name: 'mobile' },
      { width: 768, height: 1024, name: 'tablet' },
      { width: 1440, height: 900, name: 'desktop' },
    ];

    viewports.forEach((viewport) => {
      it(`matches snapshot at ${viewport.name} viewport`, async () => {
        Object.defineProperty(window, 'innerWidth', { value: viewport.width });
        Object.defineProperty(window, 'innerHeight', { value: viewport.height });

        const { container } = render(
          <FocusCoachCard currentTask="Test Task" onEmotionChange={mockOnEmotionChange} />
        );

        // Wait for any animations or state updates to complete
        await act(async () => {
          await new Promise((resolve) => setTimeout(resolve, 100));
        });

        expect(container).toMatchSnapshot(`FocusCoachCard-${viewport.name}`);
      });
    });
  });
});
