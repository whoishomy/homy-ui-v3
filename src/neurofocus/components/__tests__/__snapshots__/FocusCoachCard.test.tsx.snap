// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FocusCoachCard Visual Regression Tests matches snapshot at desktop viewport: FocusCoachCard-desktop 1`] = `
<div>
  <div
    class="rounded-lg border text-card-foreground shadow-sm relative overflow-hidden p-6 bg-gradient-to-br from-blue-50 to-white"
  >
    <div
      class="flex items-center justify-between mb-6"
    >
      <h3
        class="text-xl font-semibold text-blue-900"
      >
        Eren'in Koçu
      </h3>
      <span
        class="inline-flex items-center px-2.5 py-0.5 rounded-full font-medium border border-gray-200 text-gray-700 text-sm"
      >
        Test Task
      </span>
    </div>
    <div
      class="mt-6 grid grid-cols-2 gap-2"
    >
      <button
        class="px-4 py-2 bg-green-100 text-green-700 rounded-lg hover:bg-green-200 transition-colors"
      >
        Yapabilirim! 💪
      </button>
      <button
        class="px-4 py-2 bg-blue-100 text-blue-700 rounded-lg hover:bg-blue-200 transition-colors"
      >
        Yardım İstiyorum 🤝
      </button>
    </div>
  </div>
</div>
`;

exports[`FocusCoachCard Visual Regression Tests matches snapshot at mobile viewport: FocusCoachCard-mobile 1`] = `
<div>
  <div
    class="rounded-lg border text-card-foreground shadow-sm relative overflow-hidden p-6 bg-gradient-to-br from-blue-50 to-white"
  >
    <div
      class="flex items-center justify-between mb-6"
    >
      <h3
        class="text-xl font-semibold text-blue-900"
      >
        Eren'in Koçu
      </h3>
      <span
        class="inline-flex items-center px-2.5 py-0.5 rounded-full font-medium border border-gray-200 text-gray-700 text-sm"
      >
        Test Task
      </span>
    </div>
    <div
      class="mt-6 grid grid-cols-2 gap-2"
    >
      <button
        class="px-4 py-2 bg-green-100 text-green-700 rounded-lg hover:bg-green-200 transition-colors"
      >
        Yapabilirim! 💪
      </button>
      <button
        class="px-4 py-2 bg-blue-100 text-blue-700 rounded-lg hover:bg-blue-200 transition-colors"
      >
        Yardım İstiyorum 🤝
      </button>
    </div>
  </div>
</div>
`;

exports[`FocusCoachCard Visual Regression Tests matches snapshot at tablet viewport: FocusCoachCard-tablet 1`] = `
<div>
  <div
    class="rounded-lg border text-card-foreground shadow-sm relative overflow-hidden p-6 bg-gradient-to-br from-blue-50 to-white"
  >
    <div
      class="flex items-center justify-between mb-6"
    >
      <h3
        class="text-xl font-semibold text-blue-900"
      >
        Eren'in Koçu
      </h3>
      <span
        class="inline-flex items-center px-2.5 py-0.5 rounded-full font-medium border border-gray-200 text-gray-700 text-sm"
      >
        Test Task
      </span>
    </div>
    <div
      class="mt-6 grid grid-cols-2 gap-2"
    >
      <button
        class="px-4 py-2 bg-green-100 text-green-700 rounded-lg hover:bg-green-200 transition-colors"
      >
        Yapabilirim! 💪
      </button>
      <button
        class="px-4 py-2 bg-blue-100 text-blue-700 rounded-lg hover:bg-blue-200 transition-colors"
      >
        Yardım İstiyorum 🤝
      </button>
    </div>
  </div>
</div>
`;

exports[`FocusCoachCard renders correctly 1`] = `
<div>
  <div
    class="rounded-lg border text-card-foreground shadow-sm relative overflow-hidden p-6 bg-gradient-to-br from-blue-50 to-white"
  >
    <div
      class="flex items-center justify-between mb-6"
    >
      <h3
        class="text-xl font-semibold text-blue-900"
      >
        Eren'in Koçu
      </h3>
      <span
        class="inline-flex items-center px-2.5 py-0.5 rounded-full font-medium border border-gray-200 text-gray-700 text-sm"
      >
        Test Task
      </span>
    </div>
    <div
      class="mt-6 grid grid-cols-2 gap-2"
    >
      <button
        class="px-4 py-2 bg-green-100 text-green-700 rounded-lg hover:bg-green-200 transition-colors"
      >
        Yapabilirim! 💪
      </button>
      <button
        class="px-4 py-2 bg-blue-100 text-blue-700 rounded-lg hover:bg-blue-200 transition-colors"
      >
        Yardım İstiyorum 🤝
      </button>
    </div>
  </div>
</div>
`;
